lib/pbench/agent/tool_data_sink.py

```diff
@@ -1295,19 +1300,45 @@ def execute(self):
                 )
             self.logger.debug("published %s", self._to_client_channel)
 
-            for data in self._from_client_chan.fetch_json(self.logger):
-                ret_val = self._valid_data(data)
-                if ret_val is None:
+            # Begin listening for client state signals, and react/respond
+            # accordingly upon signal receipt
+            for signal in self.sig_resp.listen():
+                client = signal.publisher_id
+                action = signal.event
+                self.logger.debug(
+                    "client %r, action %r, metadata %r", client, action, signal.metadata
+                )
+                if action == "shutdown":
+                    # This simply means that a SignalExporter/Client we
+                    # were listening to has shut down, no action needed
+                    self.logger.debug("Client %s has shut down", client)
```

Comment:
I think this debug statement could be removed since the same data is logged at line 1308.